modules/product/src/main/java/com/opengamma/strata/product/cms/Cms.java

```diff
@@ -0,0 +1,427 @@
+/**
+ * Copyright (C) 2015 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.product.cms;
+
+import java.io.Serializable;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.Set;
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.ImmutableValidator;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.product.swap.SwapIndex;
+import com.opengamma.strata.product.swap.SwapLeg;
+
+/**
+ * The class defines a constant maturity swap (CMS) or CMS cap/floor. 
```

Comment:
Better to avoid phrases like "The class defines" at the start of the docs. Can just use "A constant maturity ...".